src/jemalloc.c

```diff
@@ -3145,13 +3157,9 @@ je_mallocx(size_t size, int flags) {
 	dopts.item_size = size;
 	if (unlikely(flags != 0)) {
 		dopts.alignment = MALLOCX_ALIGN_GET(flags);
-
 		dopts.zero = MALLOCX_ZERO_GET(flags);
-
 		dopts.tcache_ind = mallocx_tcache_get(flags);
-
-		if ((flags & MALLOCX_ARENA_MASK) != 0)
-			dopts.arena_ind = MALLOCX_ARENA_GET(flags);
+		dopts.arena_ind = mallocx_arena_get(flags);
```

Comment:
Likewise, the branch prediction in `mallocx_arena_get()` is an overkill to some extent, but there should be no harm.